code/modules/atmospherics/machinery/components/fusion/hfr_main_processes.dm

```diff
@@ -39,13 +39,13 @@
 
 	critical_threshold_proximity_archived = critical_threshold_proximity
 	if(power_level == 6)
-		critical_threshold_proximity = max(critical_threshold_proximity + max((round((internal_fusion.total_moles() * 9e5 + internal_fusion.temperature) / 9e5, 1) - 2700) / 400, 0), 0)
+		critical_threshold_proximity = max(critical_threshold_proximity + max((round((internal_fusion.total_moles() * 9e5 + coolant_temperature) / 9e5, 1) - 2700) / 400, 0), 0)
 
 	if(internal_fusion.total_moles() < 2000 || power_level <= 5)
 		critical_threshold_proximity = max(critical_threshold_proximity + min((internal_fusion.total_moles() - 1200) / 200, 0), 0)
 
-	if(internal_fusion.total_moles() > 0 && internal_fusion.temperature < 5e5 && power_level <= 4)
-		critical_threshold_proximity = max(critical_threshold_proximity + min(log(10, internal_fusion.temperature) - 5, 0), 0)
+	if(internal_fusion.total_moles() > 0 && (airs[1].total_moles() && coolant_temperature < 5e5) || power_level <= 4)
+		critical_threshold_proximity = max(critical_threshold_proximity + min(log(10, coolant_temperature + 1) - 5, 0), 0)
```

Comment:
why the + 1?

Reply:
since you can run the machine without putting in coolant, the coolant_temperature that will result will be 0, log(0) runtimes, so i add a 1 to avoid that runtime (since log(1) result in 0 there are no issues)

Reply:
couldn't you max at 1 rather then adding one? it's close to the same behavior, but it gets the idea across better